external-reports/sensor/src/test/java/org/sonarsource/plugins/externalreport/checkstyle/CheckstyleSensorTest.java

```diff
@@ -0,0 +1,235 @@
+    assertThat(first.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(first.severity()).isEqualTo(Severity.MINOR);
+    assertThat(first.remediationEffort().longValue()).isEqualTo(30L);
+    assertThat(first.primaryLocation().message()).isEqualTo("Missing package-info.java file.");
+    assertThat(first.primaryLocation().textRange()).isNull();
+
+    ExternalIssue second = externalIssues.get(1);
+    assertThat(second.primaryLocation().inputComponent().key()).isEqualTo("checkstyle-project:Main.java");
+    assertThat(second.ruleKey().rule()).isEqualTo("modifier.ModifierOrderCheck");
+    assertThat(second.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(second.severity()).isEqualTo(Severity.MINOR);
+    assertThat(second.remediationEffort().longValue()).isEqualTo(10L);
+    assertThat(second.primaryLocation().message()).isEqualTo("'static' modifier out of order with the JLS suggestions.");
+    assertThat(second.primaryLocation().textRange().start().line()).isEqualTo(2);
+
+    ExternalIssue third = externalIssues.get(2);
+    assertThat(third.primaryLocation().inputComponent().key()).isEqualTo("checkstyle-project:A.java");
+    assertThat(third.ruleKey().rule()).isEqualTo("javadoc.JavadocTypeCheck");
+    assertThat(third.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(third.severity()).isEqualTo(Severity.MAJOR);
+    assertThat(second.remediationEffort().longValue()).isEqualTo(10L);
+    assertThat(third.primaryLocation().message()).isEqualTo("Missing a Javadoc comment.");
+    assertThat(third.primaryLocation().textRange().start().line()).isEqualTo(1);
+
+    assertThat(logTester.logs(LoggerLevel.ERROR)).isEmpty();
+  }
+
+  @Test
+  public void no_issues_without_report_paths_property() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 2, null);
+    assertThat(externalIssues).isEmpty();
+    assertNoErrorWarnDebugLogs(logTester);
+  }
+
+  @Test
+  public void no_issues_with_invalid_report_path() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 2, "invalid-path.txt");
+    assertThat(externalIssues).isEmpty();
+    assertThat(onlyOneLogElement(logTester.logs(LoggerLevel.ERROR)))
+      .startsWith("FileNotFoundException:")
+      .contains("invalid-path.txt (No such file or directory)")
+      .endsWith("invalid-path.txt' can't be read.");
+  }
+
+  @Test
+  public void no_issues_with_invalid_checkstyle_file() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 2, "not-checkstyle-file.xml");
+    assertThat(externalIssues).isEmpty();
+    assertThat(onlyOneLogElement(logTester.logs(LoggerLevel.ERROR)))
+      .startsWith("IOException: Unexpected document root 'html' instead of 'checkstyle'.")
+      .endsWith("not-checkstyle-file.xml' can't be read.");
+  }
+
+  @Test
+  public void no_issues_with_invalid_line_number() throws IOException {
```

Comment:
This test is about a line number which is not a number. What about cases where the checkstyle report references a line number which is higher than the number of lines in the java file? What should we do in such cases?

Reply:
In unit tests, reporting issue on a not existing line does not fail. How to test it without having an IT?